### PR TITLE
Displaying Compound model expressions when printing compound model instances

### PR DESCRIPTION
**Pull request by anchitjain1234**
Fix for #4414 .
Example
```
In [23]:  g1 = models.Gaussian1D(1, 0, 0.2)

In [24]:  g2 = models.Gaussian1D(2.5, 0.5, 0.1)

In [25]:  gg = g1 + g2

In [26]:  print(gg)
Model: CompoundModel0
Inputs: (u'x',)
Outputs: (u'y',)
Model set size: 1
Expression: [0] + [1]
Components: 
    [0]: <Gaussian1D(amplitude=1.0, mean=0.0, stddev=0.2)>

    [1]: <Gaussian1D(amplitude=2.5, mean=0.5, stddev=0.1)>
Parameters:
    amplitude_0 mean_0 stddev_0 amplitude_1 mean_1 stddev_1
    ----------- ------ -------- ----------- ------ --------
            1.0    0.0      0.2         2.5    0.5      0.1
```

### CODE REVIEW

**Comment by embray** [comment]
@anchitjain1234 This looks great.  I have a few minor nitpicks with how I'd like the code refactored, and I'll get back to you in a bit with that.  But otherwise this looks correct.

**Comment by embray** [comment]
(Didn't mean to close)

File: astropy/modeling/core.py [hunk 1]
```diff
@@ -1874,11 +1874,7 @@ def __getattr__(cls, attr):
 
         raise AttributeError(attr)
 
-    def __repr__(cls):
-        if cls._tree is None:
-            # This case is mostly for debugging purposes
-            return cls._format_cls_repr()
-
+    def _get_expr_comp(cls):
```
Comment (embray) [inline astropy/modeling/core.py]: Instead, just call this `_format_components` and have it return the components part--not the list of `keywords` tuples.  Leave the part that builds the list of `keywords` in `__repr__` where it was.  Same in `__str__` below.  In some sense that actually has a little bit *more* code duplication, but uses the interfaces more clearly I think, and will be easier to extend later if needed.

Also move this method lower in the class--I've tried to consistently keep all "internal" methods toward the end of classes in this module.

File: astropy/modeling/core.py [hunk 2]
```diff
@@ -2433,6 +2436,10 @@ class _CompoundModel(Model):
 
     _submodels = None
 
+    def __str__(self):
+        keywords=self._get_expr_comp()
+        return super(_CompoundModel,self)._format_str(keywords=keywords)
```
Comment (embray) [inline astropy/modeling/core.py]: Should be a space after the comma here.

File: CHANGES.rst [hunk 3]
```diff
@@ -150,6 +150,9 @@ Bug fixes
 
 - ``astropy.modeling``
 
+  - Displaying Compound model expressions when printing compound model instances
+    . [#4414]
```
Comment (embray) [inline CHANGES.rst]: This is a sentence fragment.  It should be something like
"Fixed display of compound model expressions and components when printing compound model instances"

**Comment by embray** [comment]
All of the above are minor nitpicks that should be addressed.  Thanks @anchitjain1234 :+1:

File: astropy/modeling/core.py (hunk 1 above)
Comment (anchitjain1234) [inline astropy/modeling/core.py]: > and have it return the components part

How to return componets? By appending them in an empty list?

**Comment by anchitjain1234** [comment]
Is this OK @embray ?

**Comment by embray** [comment]
Perfect, thanks!

**Pull request merged**